src/values.test.ts

```diff
@@ -53,6 +53,13 @@ describe("typing", () => {
     expectTypeOf(result).toEqualTypeOf<Array<boolean>>();
   });
 
+  it("should correctly type union of Records", () => {
+    const result = values<Record<number, string> | Record<string, string>>({
+      a: "b",
+    });
```

Comment:
Don't use an explicit generic of `values`, define the type for the data itself.
```suggestion
    const result = values({
      a: "b",
    } as Record<number, string> | Record<string, string>);
```